### PR TITLE
Fix Leap, RootDelay and RootDispersion, add ReferenceTime

### PR DESCRIPTION
**Pull request by darkk**
Most of details are saved in commit messages as well :-)

### CODE REVIEW

File: ntp.go [hunk 1]
```diff
@@ -72,8 +72,9 @@ type ntpTimeShort uint32
 // Duration interprets the fixed-point ntpTimeShort as a number of elapsed
 // seconds and returns the corresponding time.Duration value.
 func (t ntpTimeShort) Duration() time.Duration {
-	sec := (t >> 16) * nanoPerSec
-	frac := (t & 0xffff) * nanoPerSec >> 16
+	t64 := uint64(t)
```
Comment (beevik) [inline ntp.go]: not sure I understand why this is necessary.  ntpTimeShort is a type alias for uint64.

File: ntp.go [hunk 2]
```diff
@@ -139,7 +143,7 @@ func Query(host string, version int) (*Response, error) {
 		ReferenceID:    m.ReferenceID,
 		RootDelay:      m.RootDelay.Duration(),
 		RootDispersion: m.RootDispersion.Duration(),
-		Leap:           (m.LiVnMode & 0x3f) >> 6,
+		Leap:           (m.LiVnMode >> 6) & 0x03,
```
Comment (beevik) [inline ntp.go]: good catch. my mask should have been `0xff`

File: ntp.go [hunk 3]
```diff
@@ -31,6 +31,10 @@ const (
 const (
 	maxStratum = 16
 	nanoPerSec = 1000000000
+	leapNoWarning = 0
+	leapAddSecond = 1
+	leapDelSecond = 2
+	leapNotInSync = 3
```
Comment (beevik) [inline ntp.go]: Nothing uses these constants, so let's remove them.

**Comment by beevik** [comment]
Thanks for the bug fix and updates.  Could you add yourself to the contributors file?

File: ntp.go (hunk 3 above)
Comment (darkk) [inline ntp.go]: I suggest to have these public constants as a part of "API" saying that Response.Leap is not {-1,0,1} but leapNoWarning / leapAddSecond / leapDelSecond / leapDelSecond. What can be better way to express that?

Comment (darkk) [inline ntp.go]: Ah, sorry, I missed that exported names should be capitalized :)

**Pull request merged**